light_curves/lc_classifier.ipynb

```diff
@@ -0,0 +1,1029 @@
+    "    which goes into the ML algorithms.\n",
+    "- chose mean and not median because there are some objects where the median flux = 0.0\n",
+    "    - if we did this before the reindexing, the median might be a non-zero value\n",
+    "- normalizing is required so that the CLAGN and it's comparison SDSS sample don't have different flux levels."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#what does the dataframe currently look like\n",
+    "df_reindex"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#idea here is that we normalize across each object.  \n",
+    "#so the code will know for example that within one object W1 will be brighter than ZTF bands\n",
+    "#but from one object to the next, it will not know that one is brighter than the other."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "#want to normalize the light curves across all bands\n",
+    "\n",
+    "#setup new column to hold the normalized fluxes\n",
+    "df_reindex[[\"normflux\"]] = 100\n",
+    "\n",
+    "#can't figure out how to do this when using multiindex, so get rid of it.\n",
+    "#remove the index\n",
+    "df_reindex_reset = df_reindex.reset_index()\n",
+    "\n",
+    "\n",
+    "#how many objects are in the dataframe?\n",
+    "#actually this is the max and some of them are missing\n",
+    "nobj = df_reindex.index.unique('objectid').max() + 1\n",
+    "\n",
+    "#for each objct in the dataframe\n",
+    "for oid in tqdm(range(nobj)):\n",
+    "    try:\n",
+    "        #what is the median flux of all bands for this object?\n",
+    "        meanflux = df_reindex.loc[oid,:,:,:].flux.mean(skipna = True)\n",
+    "        print('mean flux', meanflux)\n",
+    "    except KeyError:\n",
+    "        #some of the data cleaning above must have removed some objects\n",
+    "        print(oid, ' objectid doesnt exist')\n",
+    "    else:\n",
+    "        querystring = f'objectid == {oid}'\n",
+    "        normflux = df_reindex_reset.query(querystring).flux.values / meanflux\n",
+    "        datetime = df_reindex_reset.query(querystring).datetime.values\n",
+    "        objectid = df_reindex_reset.query(querystring).objectid.values\n",
+    "        band = df_reindex_reset.query(querystring).band.values\n",
+    "        label = df_reindex_reset.query(querystring).label.values\n",
+    "        dfsingle = pd.DataFrame(dict(flux=normflux, err=df_reindex.loc[oid,:,:,:].err, datetime=datetime, objectid=objectid, band=band,label=label))\n",
+    "\n",
+    "        #now concatenate the single new df with df_lc\n",
+    "        if oid == 0:\n",
+    "            df_norm = dfsingle\n",
+    "        else:\n",
+    "            df_norm = pd.concat([df_norm, dfsingle])\n",
```

Comment:
I changed the for loop and try/except statement to not have to worry about the objectid0 concern, but didn't work on the dfsingles and concat because this cell is likely going in the trash. I did the for loop just because maybe muscle memory will kick in on these loops and I will write them better on the first try one day (and I made a note to myself to check all of these in a future PR).   

I certainly need to fix the normalization, but will also save that for a different PR. My plan for the normalization is to write a new function which uses whatever Shooby used in her code since she has shown that she can correctly normalize such that her ML algorithm doesn't just find the different in brightness between the samples.